### PR TITLE
[curl] Delete bin in dynamic builds in Linux

### PR DESCRIPTION
**Pull request by hellozee**
- What does your PR fix?
Deletes the `bin/` directory in Linux for any type of builds. 

- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested on x64-linux with the following triplet,
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)
set(VCPKG_BUILD_TYPE release)
set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```
No, didn't update CI Baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -87,7 +87,7 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/CURL)
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (NancyLi1013) [inline ports/curl/portfile.cmake:90]: As vcpkg currently doesn't support dynamic build on Linux. It should be not necessary to add this now.

Comment (hellozee) [reply ports/curl/portfile.cmake:90]: Ahh, okay I would close this then for now, 😸

Comment (NancyLi1013) [reply ports/curl/portfile.cmake:90]: Thanks for your update. It might be needed in the near future but not for now.

Comment (hellozee) [reply ports/curl/portfile.cmake:90]: Cool

**Pull request closed without merging**